### ISSUE TITLE
Support for enums(?)

### ISSUE THREAD

**Issue by td0m** [jschaf/pggen#2]
Schema:
```sql
CREATE TYPE device_type AS ENUM ('undefined', 'phone', 'laptop', 'ipad', 'desktop', 'iot');

CREATE TABLE IF NOT EXISTS device (
  mac MACADDR PRIMARY KEY,
  type device_type NOT NULL DEFAULT 'undefined'
);
```

Queries:
```sql
-- name: List :many
SELECT *
FROM device;
```

Error:
```
infer typed named query List: infer output types for query: fetch oid types: did not find all OIDs; missing OID 16393
```

Will `pggen` support enums? If possible, it might be nice to generate go enums too, [here is how sqlc does it](https://github.com/kyleconroy/sqlc/blob/master/docs/enums.md).

**Comment by jschaf** [comment]
Yes, enums are on the roadmap.

The most correct approach is to use `pgtype.NewEnum` (maybe with a better wrapper) for nullable types. For non-nullable types, I'll can emulate what sqlc does and use a custom string type.

Alternately, I could just use a string. That downside is that this approach assumes no one will ever use the empty string as an enum value

**Comment by jschaf** [comment]
> I also discovered that pggen fails when using a nested select:

Interesting, I think that's because pgtype doesn't register the Macaddr array OID: https://github.com/jackc/pgtype/blob/v1.6.2/macaddr_array.go#L15. We can do the mapping in pggen as a workaround.  The issue isn't the select, it's that pggen doesn't know how deserialize the macaddr array type. Array types in Postgres are distinct from their element type.

**Comment by jschaf** [comment]
The MacaddrArray shoul be fixed on latest by f87c856a049c538510dec662b6f7f0366940923d.

**Comment by jschaf** [comment]
I've pushed initial enum support. It should "just work" but there's probably a bug or two lurking about.  Please open a new issue with any problems you run into.
